### PR TITLE
Test environment setup

### PR DESCRIPTION
**Pull request by GergKllai1**
# PT Link

[Test Environment Set up](https://www.pivotaltracker.com/story/show/163516827)

# Description of PR

Sets up the test environment, PR template and editor config file

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -1,7 +1,6 @@
 require_relative 'boot'
```
Comment (schneiderman805) [inline config/application.rb]: here we have single quotes and the rest are double

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
+root = true
+
+
```
Comment (HannaSwDn) [inline .editorconfig]: remove this empty line so we only have one between the code blocks

File: Gemfile [hunk 3]
```diff
@@ -2,56 +2,35 @@ source 'https://rubygems.org'
...
 
 group :development do
-  # Access an interactive console on exception pages or by calling 'console' anywhere in the code.
   gem 'web-console', '>= 3.3.0'
   gem 'listen', '>= 3.0.5', '< 3.2'
-  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem 'spring'
   gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'rb-readline', '~> 0.5.3'
 end
```
Comment (HannaSwDn) [inline Gemfile]: remove this line too

File: config/cucumber.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<%
+rerun = File.file?('rerun.txt') ? IO.read('rerun.txt') : ""
```
Comment (schneiderman805) [inline config/cucumber.yml:2]: should these be single quotes?

File: lib/tasks/cucumber.rake [hunk 5]
```diff
@@ -0,0 +1,76 @@
+# IMPORTANT: This file is generated by cucumber-rails - edit at your own peril.
+# It is recommended to regenerate this file in the future when you upgrade to a
+# newer version of cucumber-rails. Consider adding your own code to a new file
+# instead of editing this one. Cucumber will automatically load all features/**/*.rb
+# files.
+
```
Comment (HannaSwDn) [inline lib/tasks/cucumber.rake]: remove this line please

File: spec/rails_helper.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+ENV['RAILS_ENV'] ||= 'test'
+require File.expand_path('../../config/environment', __FILE__)
+abort("The Rails environment is running in production mode!") if Rails.env.production?
+require 'rspec/rails'
+begin
+  ActiveRecord::Migration.maintain_test_schema!
+rescue ActiveRecord::PendingMigrationError => e
+  puts e.to_s.strip
+  exit 1
+end
```
Comment (HannaSwDn) [inline spec/rails_helper.rb:14]: can you put space after this line please

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+require 'coveralls'
+Coveralls.wear_merged!('rails')
+ENV['RAILS_ENV'] ||= 'test'
+require File.expand_path('../../config/environment', __FILE__)
+abort("The Rails environment is running in production mode!") if Rails.env.production?
+require 'rspec/rails'
```
Comment (HannaSwDn) [inline spec/rails_helper.rb:7]: can you put space after this line please

File: spec/spec_helper.rb [hunk 8]
```diff
@@ -0,0 +1,96 @@
+# This file was generated by the `rails generate rspec:install` command. Conventionally, all
```
Comment (HannaSwDn) [inline spec/spec_helper.rb]: Remove all these comments please

File: features/support/env.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+require 'cucumber/rails'
+require 'coveralls'
+Coveralls.wear_merged!('rails')
+
+ActionController::Base.allow_rescue = false
+
+begin
+  DatabaseCleaner.strategy = :transaction
+rescue NameError
+  raise "You need to add database_cleaner to your Gemfile (in the :test group) if you wish to use it."
```
Comment (schneiderman805) [inline features/support/env.rb:10]: here we have double quotes

**Review by HannaSwDn — COMMENTED** [review]
Left comments!

**Review by schneiderman805 — CHANGES_REQUESTED** [review]
some stuff that needs to be changed

File: .editorconfig (hunk 2 above)
Comment (GergKllai1) [reply .editorconfig]: Done

File: Gemfile (hunk 3 above)
Comment (GergKllai1) [reply Gemfile]: Done

File: lib/tasks/cucumber.rake (hunk 5 above)
Comment (GergKllai1) [reply lib/tasks/cucumber.rake]: Done

File: spec/rails_helper.rb (hunk 6 above)
Comment (GergKllai1) [reply spec/rails_helper.rb:14]: Done

File: spec/rails_helper.rb (hunk 7 above)
Comment (GergKllai1) [reply spec/rails_helper.rb:7]: Done

File: spec/spec_helper.rb (hunk 8 above)
Comment (GergKllai1) [reply spec/spec_helper.rb]: DOne and Done

File: config/application.rb (hunk 1 above)
Comment (GergKllai1) [reply config/application.rb]: Done

File: config/cucumber.yml (hunk 4 above)
Comment (GergKllai1) [reply config/cucumber.yml:2]: It's scaffolded yml I will not touch that :)

File: features/support/env.rb (hunk 9 above)
Comment (GergKllai1) [reply features/support/env.rb:10]: It's scaffolded I would leave as it is

**Review by kianaditya — APPROVED** [review]
LGTM!

File: pull_requrest_template.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# PT Link
```
Comment (kianaditya) [inline pull_requrest_template.md]: File name has spelling error,will not work in github

**Review by kianaditya — CHANGES_REQUESTED** [review]

File: pull_requrest_template.md (hunk 10 above)
Comment (GergKllai1) [reply pull_requrest_template.md]: Corrected, thanks!

**Review by kianaditya — APPROVED** [review]

File: config/cucumber.yml (hunk 4 above)
Comment (schneiderman805) [reply config/cucumber.yml:2]: okay

File: features/support/env.rb (hunk 9 above)
Comment (schneiderman805) [reply features/support/env.rb:10]: okay

**Review by schneiderman805 — APPROVED** [review]

**Pull request merged**